demo/full/scripts/controllers/Settings.jsx

```diff
@@ -0,0 +1,259 @@
+      initialAudioBr,
+      minVideoBr,
+      minAudioBr,
+      maxVideoBr,
+      maxAudioBr,
+      limitVideoWidth,
+      throttleVideoBitrateWhenHidden,
+      stopAtEnd,
+      segmentRetry,
+      manifestRetry,
+      offlineRetry,
+      enableFastSwitching,
+      audioTrackSwitchingMode,
+      onCodecSwitch,
+      wantedBufferAhead,
+      maxBufferAhead,
+      maxBufferBehind,
+    } = this.state;
+
+    const initialSettings = {
+      initialVideoBr,
+      initialAudioBr,
+      minAudioBr,
+      minVideoBr,
+      maxVideoBr,
+      maxAudioBr,
+      limitVideoWidth,
+      throttleVideoBitrateWhenHidden,
+      stopAtEnd,
+      onInitialVideoBrInput: this.onInitialVideoBrInput,
+      onInitialAudioBrInput: this.onInitialAudioBrInput,
+      onMinAudioBrInput: this.onMinAudioBrInput,
+      onMinVideoBrInput: this.onMinVideoBrInput,
+      onMaxAudioBrInput: this.onMaxAudioBrInput,
+      onMaxVideoBrInput: this.onMaxVideoBrInput,
+      onLimitVideoWidthClick: this.onLimitVideoWidthClick,
+      onThrottleVideoBitrateWhenHiddenClick:
+        this.onThrottleVideoBitrateWhenHiddenClick,
+      onStopAtEndClick: this.onStopAtEndClick,
+    };
+
+    const networkConfig = {
+      segmentRetry,
+      manifestRetry,
+      offlineRetry,
+      onSegmentRetryInput: this.onSegmentRetryInput,
+      onManifestRetryInput: this.onManifestRetryInput,
+      onOfflineRetryInput: this.onOfflineRetryInput,
+    };
+
+    const trackSwitchModeConfig = {
+      enableFastSwitching,
+      audioTrackSwitchingMode,
+      onCodecSwitch,
+      onEnableFastSwitchingClick: this.onEnableFastSwitchingClick,
+      onAudioTrackSwitchingModeChange: this.onAudioTrackSwitchingModeChange,
+      onCodecSwitchChange: this.onCodecSwitchChange,
+    };
+
+    const buffersConfig = {
+      wantedBufferAhead,
+      maxBufferAhead,
+      maxBufferBehind,
+      onWantedBufferAheadInput: this.onWantedBufferAheadInput,
+      onMaxBufferAheadInput: this.onMaxBufferAheadInput,
+      onMaxBufferBehindInput: this.onMaxBufferBehindInput,
+    };
+
+    if (!this.props.showOptions) {
+      return null;
+    }
+
+    return (
+      <div style={{ marginTop: 5, display: "flex", }}>
+        <Option title="Playback">
+          <Playback
+            autoPlay={autoPlay}
+            manualBrSwitchingMode={manualBrSwitchingMode}
+            onAutoPlayClick={this.onAutoPlayClick}
+            onManualBrSwitchingModeChange={this.onManualBrSwitchingModeChange}
+          />
+        </Option>
+        <Option title="Initial settings">
+          <InitSettings {...initialSettings} />
+        </Option>
+        <Option title="Network Config">
+          <NetworkConfig {...networkConfig} />
+        </Option>
+        <Option title="Track Switch Mode">
+          <TrackSwitch {...trackSwitchModeConfig} />
+        </Option>
+        <Option title="Buffer Options">
+          <BufferOptions {...buffersConfig} />
+        </Option>
+      </div>
```

Comment:
I think that it might be better visually as two rows of three columns:
  1. This would let much more space for the option names
  2. We would not need a huge `Initial settings`

Your thoughts?

Reply:
Would you see something like this?

```jsx
<Option title="Buffer Options">
  <BufferOptions
    wantedBufferAhead={wantedBufferAhead} maxBufferAhead={maxBufferAhead} maxBufferBehind={maxBufferBehind}
    onWantedBufferAheadInput={this.onWantedBufferAheadInput} onMaxBufferAheadInput={this.onMaxBufferAheadInput}
    onMaxBufferBehindInput={this.onMaxBufferBehindInput}
  />
</Option>
```

I find it more readable with an object and spreading the props inside the component.

Reply:
No I'm thinking more generallt that instead of having 5 columns (playback, initial settings...) in one row, I guess it might be better as two rows of three columns (with the huge initial settings splitted in twos).